architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+# Radius on Dapr
+
+* **Status**: Pending/Approved
+* **Author**: Ryan Nowak (`@rynowak`)
+
+## Overview
+
+The design doc proposes replacing some existing plumbing in Radius with [Dapr](https://dapr.io) and taking an install-time dependency on Dapr. Radius needs a database and a message queue to do its work. Dapr provides implementations of its [building block APIs](https://docs.dapr.io/concepts/building-blocks-concept/) that work across self-hostable open-source technology and managed cloud services. Compared to developing our own data layer, adopting Dapr will accellerate development and production-readiness of Radius because Dapr is already mature and production-hardened. Dapr provides SDKs for most languages, including Go and C# so it can be used in the Deployment Engine.
```

Comment:
Are we replacing **etcd** and **in-memory queue**?

Reply:
We can also use the secret store of **Dapr**, I guess...

Reply:
> Are we replacing **etcd** and **in-memory queue**?

Probably yes to this. I think it's better if we can rely on Dapr and not have additional dependencies in our code.

> We can also use the secret store of Dapr, I guess...

The secret store of Dapr is **read only**. Most of our secret operations are read/write so it's not a great fit unfortunately.